src/spartan_config.erl

```diff
@@ -31,17 +31,29 @@ udp_port() ->
 bind_interface() ->
     application:get_env(?APP, bind_interface, undefined).
 
+-spec(bind_ip_blacklist() -> [inet:ipv4_address()]).
+bind_ip_blacklist() ->
+    IpStrs = application:get_env(?APP, bind_ip_blacklist, []),
+    lists:map(fun spartan_app:parse_ipv4_address/1, IpStrs).
+
+-spec(bind_ips() -> [inet:ipv4_address()]).
 bind_ips() ->
     case application:get_env(?APP, bind_ips, []) of
         [] ->
             IPs0 = bind_ips2(),
-            IPs1 = lists:usort(IPs0),
-            application:set_env(?APP, bind_ips, IPs1),
-            IPs1;
+            lager:info("found ips: ~p", [IPs0]),
+            IPs1 = bind_ip_blacklist(),
```

Comment:
As written, the blacklisted IPs will be ignored if bind_ips is set which could produce unexpected results. Here's a slightly different implementation that might be more readable - it also addresses @GoelDeepak's comment:

```
-spec(bind_ip_blacklist() -> [inet:ipv4_address()]).
bind_ip_blacklist() ->
    IpStrs = application:get_env(?APP, bind_ip_blacklist, []),
    lists:map(fun spartan_app:parse_ipv4_address/1, IpStrs).

-spec(bind_ips() -> [inet:ipv4_address()]).
bind_ips() ->
    IPs = case application:get_env(?APP, bind_ips, []) of
        [] ->
            DefaultIPs = default_bind_ips(),
            application:set_env(?APP, bind_ips, DefaultIPs),
            DefaultIPs;
        V ->
            V
    end,
    BlacklistedIPs = bind_ip_blacklist(),
    [ IP || IP <- IPs0, not lists:member(IP, BlacklistedIPs) ],
    

-spec(default_bind_ips() -> [inet:ipv4_address()]).
default_bind_ips() ->
    IFs0 = get_ip_interfaces(),
    IPs = case bind_interface() of
        undefined ->
            [Addr || {_IfName, Addr} <- IFs0];
        ConfigInterfaceName ->
            IFs1 = lists:filter(fun({IfName, _Addr}) -> string:equal(IfName, ConfigInterfaceName) end, IFs0),
            [Addr || {_IfName, Addr} <- IFs1]
    end,
    lists:usort(IPs).
```